src/tests/system/tests/test_identity.py

```diff
@@ -496,3 +496,42 @@ def test_identity__lookup_users_fully_qualified_name_and_case_insensitive(client
         result = client.tools.id(name)
         assert result is not None, f"User {name} was not found using id"
         assert result.memberof([103, 1003]), f"User {name} is member of wrong groups"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.authentication
+@pytest.mark.topology(KnownTopologyGroup.AnyAD)
+def test_identity__lookup_user_posix_and_non_posix(client: Client, provider: GenericADProvider):
+    """
+    :title: with ldap provider idmapping is disabled
+    :setup:
+        1. Configure ldap_provider with idmapping disabled
+    :steps:
+        1. Fetch non-posix users and groups information
+        2. Fetch posix users and groups information
+        3. Log in as a posix-user
+    :expectedresults:
+        1. Non-posix user and group should not be returned
+        2. POSIX user and group information should be returned
+        3. POSIX user should be able to log in
+    :customerscenario: False
+    """
+
+    u1 = provider.user("posix_user").add(uid=10001, gid=20001, password="Secret123",
+                                         gecos='User for tests',
+                                         shell='/bin/bash')
+    provider.group("posix_group").add(gid=20001).add_member(u1)
+
+    u2 = provider.user("nonposix_user").add(password="Secret123")
+    provider.group("nonposix_group").add().add_member(u2)
+
+    client.sssd.domain["ldap_id_mapping"] = "false"
+    client.sssd.start()
+
+    assert client.auth.ssh.password("posix_user", "Secret123"), 'ssh login failed for posix-user'
```

Comment:
We should ssh as nonposix_user as well but "assert not " expecting a failure. Honestly, this is kind of redundant. Let's remove this actually, the login.

Reply:
removed.